src/client.rs

```diff
@@ -610,13 +619,18 @@ where
 
                     // Admin clients ignore shutdown.
                     else {
-                        read_message(&mut self.read).await?
+                        read_message(&mut self.read, &mut self.client_message_buffer).await?
```

Comment:
I _think_ we should rename `read_message` to `read_message_into_buffer` now. the old implementation returned a buffer, and now it writes values into the buffer. i don't have a strong opinion on this, but @levkk what do you think?

Reply:
It would be nice to keep both for different use cases, I agree.